### DISCUSSION TITLE
Not only corners but also the lines

### DISCUSSION THREAD

**Discussion by Ron0x99** [GriefPrevention/GriefPrevention#823] [Ideas]
I wanted to ask if you could add a way that you can see the outlines of the complete claim. We have people building underground and they have a difficult time seeing the area of their claimed land underground.  But if it would just show the lines from one corner to the next it would be so much easier (but also down so you can see the borders of your claim). Something like what WorldEditSUI does for the WorldEdit plugin.

**Comment by RoboMWM** [comment]
CUI is client side afaik.

There is a plugin that uses structure blocks but it's very limited - but would make sense to do in an addon

Otherwise there is no other way to show the lines unfortunately. Claim visualization is one of the things that's been on my mind a lot, with still no ideal way to figure out how to best present it to a user who is underground.

**Comment by shai** [comment]
> Otherwise there is no other way to show the lines unfortunately. Claim visualization is one of the things that's been on my mind a lot, with still no ideal way to figure out how to best present it to a user who is underground.

What about showing the claim area above ground? Even /claimslist only shows the first corner of the claim (XZ) and doesn't show the entire claim nor does it give the opposite side cords of the claim.

**Comment by RoboMWM** [comment]
I'm confused what you're trying to ask or justify there...

**Comment by shai** [comment]
Please accept my apologies. I didn't mean to confuse anyone.
What I'm actually asking is, how do I go about showing myself (or the user using the plugin) what my claim borders are, right now.

**Comment by RoboMWM** [comment]
right click with stick.

Gonna hide your comments since it's off-topic from this one fyi.

**Comment by Jikoo** [comment]
Structure blocks have a very small size and CUI requires client modification. Perhaps clientside glowing falling blocks? GP does already draw borders with clientside blocks. This would allow it to render without existing downsides (fake blocks in water cause anticheat issues, things of that nature) and with the benefit of being visible through walls. Unfortunately, this would require either ProtocolLib or version-specific code. Not sure without looking whether or not it's possible for an addon to entirely overwrite GP's visualization process, but it's food for thought.
